### PR TITLE
init working version commit

### PR DESCRIPTION
**Pull request by emmalu**
Updates include:
- UI refresh
- Layers widget that includes:
-- Organized Biomass Layers with opacity control
-- Select Basemap Layers with opacity control
- STAC widget that:
-- is connected to the MAAP stac catalog (that is currently in transition so not all datasets are available as of 5/5)
-- automatically searches/matches items per selected collection
-- checks for COGs and, if valid, allows user to add data to the map
-- allows the user to change the selected color palette for rendering the selected item (color palettes have been organized into their respective categories for easier navigation)

Biomass Layers UI

Basemaps UI

STAC Widget UI

STACT Viz Palette UI

### CODE REVIEW

**Comment by emmalu** [comment]
@abarciauskas-bgse @wildintellect

The latest changes include: 
- integrating a better **Draw Tools** UX that aligns with other current design decisions (with output coordinates shown as code)
- **COG integration** of new JPL Global Above Ground Biomass layers (with the ability to update color palettes)
- **synced with main branch changes** to ensure successful install / testing from within the ADE environment

Latest Fast Browse v2 UI snapshot of JPL dataset in blues


cc @heidimok @zacharyDez @edkeeble

**Comment by emmalu** [comment]
@emileten @jjfrench @abarciauskas-bgse 
The package is ready to be installed/tested, following these docs: 
https://github.com/MAAP-Project/maap-documentation/pull/190
[View Directly in ReviewNB](https://app.reviewnb.com/MAAP-Project/maap-documentation/blob/add-stac-ipyleaflet-docs/docs/source/technical_tutorials/visualization/stac_ipyleaflet.ipynb/)

Use the branch name when installing, in order to see v2 updates:
`pip install git+https://github.com/MAAP-Project/stac_ipyleaflet.git@dev/wug-map-refresh#egg-info=stac_ipyleaflet`

cc @wildintellect

File: stac_ipyleaflet/stac_discovery/stac_widget.py [hunk 1]
```diff
@@ -154,154 +280,271 @@ def list_palettes(add_extra=False, lowercase=False):
...
+        def update_default_display_settings():
+            assets = [i for i in self.stac_data["items"] if i["id"] == items_dropdown.value][0]["assets"]
+            href = [i for i in self.stac_data["items"] if i["id"] == items_dropdown.value][0]["href"]
+                      
+            if len(assets.keys()) > 0:
+                with output:
+                    output.clear_output()
+                    # print("ASSETS", assets)
+                asset_keys = assets.keys()
+                if "data" in asset_keys:
```
Comment (jjfrench) [inline stac_ipyleaflet/stac_discovery/stac_widget.py:353]: asset names aren't always consistent with "data" or "cog_default", assets can theoretically be anything descriptive like "red", "green", "blue", or "coastal" (this is why icesat2-boreal currently isn't working 😢)

Comment (wildintellect) [reply stac_ipyleaflet/stac_discovery/stac_widget.py:353]: Even better, multiple assets might be COGs. Long term we need a better approach. I wonder if we can set a "cog_default" by reading the media types on the collection assets?
https://stac.maap-project.org/collections/icesat2-boreal

image

role = data + type = "image/tiff; application=geotiff; profile=cloud-optimized" (there are a few variants)

For now easiest thing would be to rename the "tif" asset to "data" or "cog_default"?

Comment (emmalu) [reply stac_ipyleaflet/stac_discovery/stac_widget.py]: > asset names aren't always consistent with "data" or "cog_default", assets can theoretically be anything descriptive like "red", "green", "blue", or "coastal" (this is why icesat2-boreal currently isn't working 😢)

I implemented the if "data" else "cog_default" logic when pivoting between the original AFRISAR and new JPL datasets - which both seemed to have different asset structures. Now that I'm seeing the latest boreal ice, I think I better understand that an item can have multiple assets associated. I've refactored to loop through each asset and within each, determine its media_type, looking for `geotiff` & `cloud-optimized` values to determine if it is compatible. Regardless, the boreal ice is still not displaying... 😞

**Comment by jjfrench** [comment]
The revamped UI looks ✨!  Just a few questions/comments

- The histogram feature doesn't seem to be accessible, I wasn't in the project kickoff meeting but was this discussed as something to remove? (What Draw and Layers were originally used for)
- Should users only be allowed to draw one bounding box? I'm able to draw more than one; the draw tool gets stuck on "release mouse to finish drawing" when I attempt to draw a second bbox.
- There's no way to "change opacity"/remove a STAC item - issue when trying to display a STAC item and Biomass layer in the same region
- Minor: Selecting a collection populates the items, selecting the item dropdown shows the first item selected but the dropdown value isn't set. Maybe deselect the first item in the dropdown or fill the dropdown value to the first item in the list?
- Minor: Displaying an item on the map attempts to center the map on the STAC item but often the overlay ends up behind the widget. Maybe an offset to center the map a little farther right?
- Minor: Several of the Biomass layers use zoom >= 8 but the current zoom level isn't shown.

**Review by jjfrench — APPROVED** [review]
Reviewed the widget's features - looks good for 0.2.0!

**Pull request merged**